integration/simulation/transaction_injector.go

```diff
@@ -346,13 +346,14 @@ func readNonce(cl *obscuroclient.Client, a common.Address) uint64 {
 	return result
 }
 
+// NextNonce waits for the previous transaction to be recorded - returns a value if a timeout has passed
 func NextNonce(cl *obscuroclient.Client, w wallet.Wallet) uint64 {
-	// only returns the nonce when the previous transaction was recorded
-	for {
+	for startTime := time.Now(); time.Since(startTime) < 5*time.Second; time.Sleep(time.Millisecond) {
```

Comment:
probably not best to measure this in seconds but in avgBlock Time units

Reply:
Good point 👍